src/main/java/teammates/ui/pagedata/InstructorFeedbackResultsPageData.java

```diff
@@ -1202,6 +1179,40 @@ private void buildTableColumnHeaderForRecipientQuestionGiverView(List<ElementTag
         return responseRows;
     }
 
+    private void addCommentsToResponseRow(FeedbackResponseAttributes response, FeedbackQuestionAttributes question,
+            InstructorFeedbackResultsResponseRow responseRow, Map<String, Integer> userIndexesForComments) {
+        responseRow.setCommentsOnResponsesAllowed(true);
+        String giverNameAndTeam = bundle.getNameForEmail(response.giver);
+        String recipientNameAndTeam = bundle.getNameForEmail(response.recipient);
+
+        String giverTeam = bundle.getTeamNameForEmail(response.giver);
+        String recipientTeam = bundle.getTeamNameForEmail(response.recipient);
+
+        giverNameAndTeam = bundle.appendTeamNameToName(giverNameAndTeam, giverTeam);
+        recipientNameAndTeam = bundle.appendTeamNameToName(recipientNameAndTeam, recipientTeam);
+
+        List<FeedbackResponseCommentRow> comments =
+                buildResponseComments(giverNameAndTeam, recipientNameAndTeam, question, response);
+        if (!comments.isEmpty()) {
+            responseRow.setCommentsOnResponses(comments);
+        }
+        Map<FeedbackParticipantType, Boolean> responseVisibilityMap = getResponseVisibilityMap(question);
+        FeedbackResponseCommentRow addCommentForm = buildFeedbackResponseCommentAddForm(question, response,
+                responseVisibilityMap, giverNameAndTeam, recipientNameAndTeam);
+        responseRow.setAddCommentButton(addCommentForm);
+        if (userIndexesForComments.get(response.giver) == null) {
+            userIndexesForComments.put(response.giver, userIndexesForComments.size() + 1);
+        }
+        int responseGiverIndex = userIndexesForComments.get(response.giver);
+        if (userIndexesForComments.get(response.recipient) == null) {
+            userIndexesForComments.put(response.recipient, userIndexesForComments.size() + 1);
+        }
+        int responseRecipientIndex = userIndexesForComments.get(response.recipient);
+
+        responseRow.setResponseRecipientIndex(responseRecipientIndex);
```

Comment:
As we identify a comment model by `id="commentModal-${responseRecipientIndex}-${responseGiverIndex}-${questionIndex}`. The above method won't give us a unique ID in a HTML page because the combination is not unique outside `buildResponseRowsForQuestionForSingleParticipant`. In GQR view, Image for two different givers, they will have same question displayed respectively. Each panel will result in calling of `buildResponseRowsForQuestionForSingleParticipant`. So `questionIndex` would be the same. `responseRecipientIndex` and `responseGiverIndex` could also be the same because of the way we generate ID.

Reply:
Please discuss either here or slack before go for another implementation.